storage-dynamodb/src/main/java/io/delta/storage/DynamoDBLogStore.java

```diff
@@ -16,10 +16,242 @@
 
 package io.delta.storage;
 
+import org.apache.hadoop.fs.Path;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.io.IOException;
+
 import org.apache.hadoop.conf.Configuration;
 
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ComparisonOperator;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.amazonaws.services.dynamodbv2.model.ExpectedAttributeValue;
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import com.amazonaws.services.dynamodbv2.model.ResourceInUseException;
+import com.amazonaws.services.dynamodbv2.model.ScalarAttributeType;
+import com.amazonaws.auth.AWSCredentialsProvider;
+
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+
+import java.util.NoSuchElementException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
 public class DynamoDBLogStore extends BaseExternalLogStore {
-    public DynamoDBLogStore(Configuration hadoopConf) {
+    private static final Logger LOG = LoggerFactory.getLogger(DynamoDBLogStore.class);
+
+    private AmazonDynamoDBClient client = null;
+    private String tableName;
+    private String credentialsProviderName;
+    private String regionName;
+    private String endpoint;
+
+    public DynamoDBLogStore(Configuration hadoopConf) throws IOException {
         super(hadoopConf);
+        tableName = getParam(hadoopConf, "tableName", "delta_log");
+        credentialsProviderName = getParam(
+            hadoopConf,
+            "credentials.provider",
+            "com.amazonaws.auth.DefaultAWSCredentialsProviderChain"
+        );
+        regionName = getParam(hadoopConf, "region", "us-east-1");
+        endpoint = getParam(hadoopConf, "endpoint", null);
```

Comment:
@mrk-its - why expose both `endpoint` and `region` as configuration parameters that the user can specify?

Reply:
we definitely need to expose `region` - as it specifies AWS region in which table is located.
We can probably get rid of `endpoint` - it is useful only for testing, I was using it to connect to local dynamodb instance in docker, no we are not using it.